rascal-vscode-extension/src/extension.ts

```diff
@@ -220,19 +220,21 @@ function startTerminal(uri: vscode.Uri, ...extraArgs: string[]) {
         rascalClient = activateRascalLanguageClient();
     }
     console.log("Starting:" + uri + extraArgs);
-    Promise.all([
-        rascalClient.then(c => c.sendRequest<IDEServicesConfiguration>("rascal/supplyIDEServicesConfiguration")),
-        rascalClient.then(c => c.sendRequest<string[]>("rascal/supplyProjectCompilationClasspath", { uri: uri.toString() }))
-    ]).then(async cfg => {
-        let terminal = vscode.window.createTerminal({
-            cwd: path.dirname(uri.fsPath),
-            shellPath: await getJavaExecutable(),
-            shellArgs: buildShellArgs(cfg[1], cfg[0], ...extraArgs),
-            name: 'Rascal Terminal',
-        });
+    rascalClient.then(c => c.onReady().then(() => {
+        Promise.all([
+            rascalClient!.then(c => c.sendRequest<IDEServicesConfiguration>("rascal/supplyIDEServicesConfiguration")),
```

Comment:
You could move the on ready part to this then chain.

We could also make this functions async, so we can just call them with an await and avoid al this chaining of then calls.